check_for_openmpi.cmake

```diff
@@ -0,0 +1,11 @@
+# Sets COSMA_WITH_OPENMPI to ON if OpenMPI was found and to OFF otherwise.
+#
+function(check_for_openmpi)
+  execute_process(COMMAND mpirun --version OUTPUT_VARIABLE MPIRUN_OUTPUT)
+  string(FIND "${MPIRUN_OUTPUT}" "Open MPI" OMPI_POS)
+  if(OMPI_POS STREQUAL "-1")
+    set(COSMA_WITH_OPENMPI OFF PARENT_SCOPE)
+  else()
+    set(COSMA_WITH_OPENMPI ON PARENT_SCOPE)
+  endif()
+endfunction()
```

Comment:
I think it would be better to change this script to detect the type of MPI implementation found by [FindMPI](https://cmake.org/cmake/help/v3.15/module/FindMPI.html), by introspection on variables.

It seems that [MPI_DETERMINE_LIBRARY_VERSION](https://cmake.org/cmake/help/v3.15/module/FindMPI.html#advanced-variables-for-using-mpi) may be helpful for this; if it (more or less) clearly states the implementation, I would go for embedding this detection directly into the MKL_ScaLAPACK section without having an additional cmake script for this function.